app/services/experimentation/stale_patient_selection.rb

```diff
@@ -1,27 +1,58 @@
 module Experimentation
   class StalePatientSelection
+    ELIGIBILITY_START = 365.days.freeze
+    ELIGIBILITY_END = 35.days.freeze
+
     def self.call(*args)
       new(*args).call
     end
 
-    attr_reader :eligible_range
     attr_reader :start_date
+    attr_reader :eligible_range
 
-    def initialize(start_date:, eligible_range:)
-      @eligible_range = eligible_range
+    def initialize(start_date:)
       @start_date = start_date
+      range_start = (start_date - ELIGIBILITY_START).beginning_of_day
+      range_end = (start_date - ELIGIBILITY_END).end_of_day
+      @eligible_range = (range_start..range_end)
     end
 
     def call
-      Experiment.candidate_patients.joins(:encounters)
-        .where(encounters: {device_created_at: eligible_range})
-        .where("NOT EXISTS (SELECT 1 FROM encounters WHERE encounters.patient_id = patients.id AND
-              encounters.device_created_at > ?)", eligible_range.end)
-        .left_joins(:appointments)
-        .where("NOT EXISTS (SELECT 1 FROM appointments WHERE appointments.patient_id = patients.id AND
-              appointments.scheduled_date >= ?)", start_date)
-        .distinct
-        .pluck(:id)
+      candidate_ids = Experiment.candidate_patients.pluck(:id)
+      return [] if candidate_ids.empty?
+
+      parameters = {
+        candidate_ids: candidate_ids,
+        hypertension: "yes",
+        range_start: eligible_range.begin,
+        range_end: eligible_range.end,
+        start_date: start_date
+      }
+      sql = GitHub::SQL.new(<<~SQL, parameters)
```

Comment:
I'm using SQL here to make sure we have the logic correct first.  It is easier to do these types of queries in plain SQL compared to fighting with AR complexity for things like sub selects and such.  We can always refactor to AR later if we want.

Also, we may need raw SQL if we use the table sampling method we had discussed to grab random ids.